presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestSphericalGeoFunctions.java

```diff
@@ -144,4 +150,54 @@ private void assertDistance(String wkt, String otherWkt, Double expectedDistance
     {
         assertFunction(format("ST_Distance(to_spherical_geography(ST_GeometryFromText('%s')), to_spherical_geography(ST_GeometryFromText('%s')))", wkt, otherWkt), DOUBLE, expectedDistance);
     }
+
+    @Test
+    public void testArea()
+            throws IOException
+    {
+        // Empty polygon
+        assertFunction("ST_Area(to_spherical_geography(ST_GeometryFromText('POLYGON EMPTY')))", DOUBLE, 0.0);
+
+        // Invalid polygon (too few vertices)
+        assertInvalidFunction("ST_Area(to_spherical_geography(ST_GeometryFromText('POLYGON((90 0, 0 0))')))", ".*");
```

Comment:
Let's verify error message as well, e.g. replace ".*" with "Polygon is not valid: a loop contains less then 3 vertices."